sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -73,6 +73,20 @@ static gnrc_pktsnip_t *_make_netif_hdr(uint8_t *mhr)
     return snip;
 }
 
+#if MODULE_GNRC_NETIF_DEDUP
+static inline bool _already_received(gnrc_netif_t *netif,
+                                     gnrc_netif_hdr_t *netif_hdr,
+                                     uint8_t *mhr)
```

Comment:
Seeing this makes me wonder a bit about the netif data structures, although not related to this PR...

Reply:
What exactly are you wondering about?

Reply:
About that fact that we need three "data structures"  in order to get header information such as sequence number or link layer addresses.

Reply:
> About that fact that we need three "data structures" in order to get header information such as sequence number or link layer addresses.

For that we "only" need two. The third (`netif`) is the current state against we check that information. We wouldn't actually need two either, but the addresses in the IEEE 802.15.4 are rather complicated to parse out due to its and their variable lengths, so why not actually use the results from a previous parsing step i.e. `netif_hdr` (we don't need the sequence number in that for most use-cases)